pkg/registry/core/pod/strategy.go

```diff
@@ -226,6 +226,11 @@ func (podStatusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.
 	if newPod.Status.QOSClass == "" {
 		newPod.Status.QOSClass = oldPod.Status.QOSClass
 	}
+	// If a client request tries to clear `status.observedGeneration`, we
+	// preserve the original value.
+	if newPod.Status.ObservedGeneration == 0 {
+		newPod.Status.ObservedGeneration = oldPod.Status.ObservedGeneration
+	}
 }
 
 func (podStatusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
```

Comment:
Adding a response here to the comment on https://github.com/kubernetes/kubernetes/pull/130397#pullrequestreview-2638199668:

> We need validation for the new field


When writing up the KEP I think we agreed that kubelet should be the source of truth for observedGeneration, so we will permit any updates on the API side including regressions back to decreasing values (https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/5067-pod-generation#client-requests-to-update-observedgeneration), so not sure if we actually need to add any additional validation here?

Reply:
validation would be added inside the ValidatePodStatusUpdate method

I think we would just validate these are non-negative integers, and allow any value >= 0

there's an existing ValidateNonnegativeField helper method you can use, something like `apivalidation.ValidateNonnegativeField(int64(status.ObservedGeneration), fieldPath.Child("observedGeneration"))`

validate the status and condition ObservedGeneration fields